src-java/kilda-persistence-tinkerpop/src/main/java/org/openkilda/persistence/ferma/repositories/FermaFlowRepository.java

```diff
@@ -622,4 +683,41 @@ protected void doRemove(FlowFrame frame) {
     protected FlowData doDetach(Flow entity, FlowFrame frame) {
         return Flow.FlowCloner.INSTANCE.deepCopy(frame, entity);
     }
+
+    /**
+     * Find all flows with filtering by flow IDs. This method does it as a single operation for better performance.
+     * @param flowIds Flow IDs for filtering.
+     * @return an empty collection if flowIds is empty,
+     *     otherwise returns a collection of Flows with the provided IDs if they exist.
+     */
+    private Collection<Flow> findFlowsByFlowIds(Collection<String> flowIds) {
+        if (flowIds == null || flowIds.size() == 0) {
```

Comment:
We could replace (flowIds == null || flowIds.size() == 0) with isEmpty(flowIds) method from org.apache.commons.collections.CollectionUtils or another library.